app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java

```diff
@@ -15,71 +15,92 @@
+        return R.layout.activity_search_contacts;
+    }
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        listContactsView = findViewById(R.id.listContacts);
+        searchUsers = findViewById(R.id.search_users);
+
+        loaderManager = LoaderManager.getInstance(this);
+
         initialize();
     }
 
     protected void initialize() {
-        loadContacts();
-    }
-
-    private void loadContacts() {
-        ContentResolver contentResolver = getContentResolver();
-        Cursor cursor = contentResolver.query(ContactsContract.Contacts.CONTENT_URI, null, null, null, null);
-
         list = new ArrayList<>();
+        updateList();
 
-        if (cursor.getCount() > 0) {
-            while (cursor.moveToNext()) {
-                String id = cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts._ID));
-                String name = cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
-                int hasPhoneNumber = Integer.parseInt(cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts.HAS_PHONE_NUMBER)));
+        setupContactListClick();
+        setupSearchListener();
 
-                if (hasPhoneNumber > 0) {
-                    Cursor cursor2 = contentResolver.query(ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
-                            null,
-                            ContactsContract.CommonDataKinds.Phone.CONTACT_ID + "= ?",
-                            new String[]{id}, null);
-
-                    while (cursor2.moveToNext()) {
-                        String PhoneNumber = cursor2.getString(cursor2.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
+        loaderManager.initLoader(0, null, this);
+    }
 
-                        list.add(name + "\n" + PhoneNumber);
-                    }
+    private void setupSearchListener() {
+        searchUsers.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                searchString = charSequence.toString();
+                loaderManager.restartLoader(0, null, SearchContactsToSupervise.this);
+            }
 
-                    cursor2.close();
-                }
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                // Nothing happens before user search input
             }
-        }
-        cursor.close();
 
-        listShow = list;
-        ArrayAdapter<String> adapter = new ArrayAdapter<>(getBaseContext(), android.R.layout.simple_list_item_1, listShow);
-        listContactsView = findViewById(R.id.listContacts);
-        listContactsView.setAdapter(adapter);
+            @Override
+            public void afterTextChanged(Editable editable) {
+                // Nothing happens aster user search input
```

Comment:
Probably just a typing error here, maybe you wanted to say "after"...

Reply:
good catch, will fix

Reply:
@Landry333 Fixed